src/main/java/competition/simulation/Simulator.java

```diff
@@ -0,0 +1,61 @@
+package competition.simulation;
+
+import competition.subsystems.drive.DriveSubsystem;
+import competition.subsystems.pose.PoseSubsystem;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import xbot.common.math.MovingAverageForDouble;
+import xbot.common.math.MovingAverageForTranslation2d;
+
+import javax.inject.Inject;
+
+public class Simulator {
+
+    MovingAverageForTranslation2d translationAverageCalculator =
+            new MovingAverageForTranslation2d(15);
+
+    MovingAverageForDouble rotationAverageCalculator =
+            new MovingAverageForDouble(15);
+
+    PoseSubsystem pose;
+    DriveSubsystem drive;
+
+    @Inject
+    public Simulator(PoseSubsystem pose, DriveSubsystem drive) {
+        this.pose = pose;
+        this.drive = drive;
+    }
+
+    public void update() {
+        double robotTopSpeedInMetersPerSecond = 3.0;
```

Comment:
⭐ consider pulling this up to be class level `final` variables so it's easier to see the configuration of the class